corehq/messaging/smsbackends/ivory_coast_mtn/models.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+import pytz
+import requests
+from datetime import datetime
+from corehq.apps.sms.models import SQLSMSBackend
+from corehq.messaging.smsbackends.ivory_coast_mtn.exceptions import IvoryCoastMTNError
+from corehq.messaging.smsbackends.ivory_coast_mtn.forms import IvoryCoastMTNBackendForm
+from corehq.apps.sms.models import SMS
+from corehq.apps.sms.util import strip_plus
+from corehq.util.timezones.conversions import ServerTime
+from lxml import etree
+
+
+class IvoryCoastMTNBackend(SQLSMSBackend):
+
+    class Meta:
+        app_label = 'sms'
+        proxy = True
+
+    @classmethod
+    def get_api_id(cls):
+        return 'IVORY_COAST_MTN'
+
+    @classmethod
+    def get_generic_name(cls):
+        return "Ivory Coast MTN"
+
+    @classmethod
+    def get_available_extra_fields(cls):
+        return [
+            'customer_id',
+            'username',
+            'password',
+            'sender_id',
+        ]
+
+    @classmethod
+    def get_form_class(cls):
+        return IvoryCoastMTNBackendForm
+
+    @classmethod
+    def get_ivory_coast_timestamp(cls):
+        return ServerTime(datetime.utcnow()).user_time(pytz.timezone('Africa/Abidjan')).done()
+
+    def get_params(self, msg_obj):
+        config = self.config
+
+        return {
+            'customerID': config.customer_id,
+            'userName': config.username,
+            'userPassword': config.password,
+            'originator': config.sender_id,
+            # This is for a custom project and they only send messages in French
+            'messageType': 'Latin',
+            # This must be set to the date and time to send the message; it's a required field
+            'defDate': self.get_ivory_coast_timestamp().strftime('%Y%m%d%H%M%S'),
+            'blink': 'false',
+            'flash': 'false',
+            'private': 'false',
+            'smsText': msg_obj.text.encode('utf-8'),
+            'recipientPhone': msg_obj.phone_number,
+        }
+
+    @classmethod
```

Comment:
out of curiosity, why a `@classmethod` over a `@staticmethod`?

Reply:
yeah staticmethod would be more appropriate, https://github.com/dimagi/commcare-hq/pull/18546/commits/4ba3e561dd7d9d07ee5cdaf09a0313f22bf3a47c